### PR TITLE
Move blockchain-utils into js-ceramic

### PR DESCRIPTION
**Pull request by ukstv**
Instead of #763

### CODE REVIEW

File: packages/blockchain-utils-linking/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Utils for linking blockchain accounts
+
+> This package contains utility functions to link blockchain accounts to a DID.
+
+### Installation
+```shell
+npm install @ceramicnetwork/doctype-blockchain-utils-linking
```
Comment (oed) [inline packages/blockchain-utils-linking/README.md:7]:
```suggestion
npm install @ceramicnetwork/blockchain-utils-linking
```

File: packages/blockchain-utils-validation/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Utils for validation of linked blockchain accounts
+
+> This package contains utility functions to validate links of blockchain accounts linked a DID.
+
+### Installation
+```shell
+npm install @ceramicnetwork/doctype-blockchain-utils-validation
```
Comment (oed) [inline packages/blockchain-utils-validation/README.md:7]:
```suggestion
npm install @ceramicnetwork/blockchain-utils-validation
```

File: packages/blockchain-utils-linking/src/ethereum.ts [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    hexMessage,
+    account.address,
+  ]);
+  const signature = await safeSend(payload, provider);
+  if (account) {
+    const recoveredAddr = verifyMessage(message, signature).toLowerCase();
+    if (account.address !== recoveredAddr)
+      throw new Error(
+          "Provider returned signature from different account than requested"
+      );
```
Comment (oed) [inline packages/blockchain-utils-linking/src/ethereum.ts:184]: Do we need this check really? If not we could remove the dependency on `@ethersproject/wallet` in this package.

File: packages/blockchain-utils-validation/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+const handlers = {
+  [ethereum.namespace]: ethereum,
+  [filecoin.namespace]: filecoin,
+  [polkadot.namespace]: polkadot,
+  [eosio.namespace]: eosio,
+};
+
+const findDID = (did: string): string | undefined =>
+  did.match(/(did:(3|muport):[a-zA-Z0-9])\w+/)?.[0];
```
Comment (oed) [inline packages/blockchain-utils-validation/src/index.ts:16]: Would you mind changing this as per: https://github.com/ceramicnetwork/js-3id-blockchain-utils/pull/45 🙏

**Review by oed — CHANGES_REQUESTED** [review]
Looks good, just a few changes needed.

File: packages/blockchain-utils-linking/src/ethereum.ts (hunk 3 above)
Comment (ukstv) [reply packages/blockchain-utils-linking/src/ethereum.ts:184]: Really not. I consider it an attempt to be smart and preemptively fix a (probably non-existent) problem of a provider misbehaving.

File: packages/blockchain-utils-validation/src/index.ts (hunk 4 above)
Comment (ukstv) [reply packages/blockchain-utils-validation/src/index.ts]: Done.

**Review by oed — APPROVED** [review]

**Comment by stbrody** [comment]
Removed myself as reviewer, I'm happy to defer to @oed's review here as he's much more familiar with this code than I am

**Pull request merged**